apps/methodologies/bold/rule-processors/mass/same-source-and-pick-up-addresses/src/lib/same-source-and-pick-up-addresses.processor.ts

```diff
@@ -1,90 +1,80 @@
-import { loadParentDocument } from '@carrot-fndn/methodologies/bold/io-helpers';
+import type { EvaluateResultOutput } from '@carrot-fndn/shared/rule/standard-data-processor';
+
 import {
-  and,
-  eventNameIsAnyOf,
   isActorEventWithSourceActorType,
   metadataAttributeValueIsAnyOf,
 } from '@carrot-fndn/methodologies/bold/predicates';
+import { ParentDocumentRuleProcessor } from '@carrot-fndn/methodologies/bold/processors';
 import {
+  type Document,
+  type DocumentEvent,
   DocumentEventAttributeName,
   DocumentEventMoveType,
-  DocumentEventName,
 } from '@carrot-fndn/methodologies/bold/types';
-import { DOCUMENT_NOT_FOUND_RESULT_COMMENT } from '@carrot-fndn/methodologies/bold/utils';
-import { RuleDataProcessor } from '@carrot-fndn/shared/app/types';
-import { isNonEmptyArray, toDocumentKey } from '@carrot-fndn/shared/helpers';
-import { mapToRuleOutput } from '@carrot-fndn/shared/rule/result';
-import {
-  type RuleInput,
-  type RuleOutput,
-  RuleOutputStatus,
-} from '@carrot-fndn/shared/rule/types';
+import { isNil } from '@carrot-fndn/shared/helpers';
+import { RuleOutputStatus } from '@carrot-fndn/shared/rule/types';
+
+interface RuleSubject {
+  evaluateEvent?: DocumentEvent | undefined;
+  sourceActorEvent?: DocumentEvent | undefined;
+}
 
-export class SameSourceAndPickUpAddressesProcessor extends RuleDataProcessor {
-  static resultComments = {
-    doNotMatch: 'Source and pick-up addresses do not match',
-    noSourceActorEvent: 'No actor event with source actor type found',
-    ruleNotApplicable:
+export class SameSourceAndPickUpAddressesProcessor extends ParentDocumentRuleProcessor<RuleSubject> {
+  private ResultComments = {
+    DO_NOT_MATCH: 'Source and pick-up addresses do not match',
+    NO_SOURCE_ACTOR_EVENT: 'No actor event with source actor type found',
+    RULE_NOT_APPLICABLE:
       'Rule not applicable: No move event with pick-up move type found',
```

Comment:
Acho que teria que ser algo como:
```suggestion
      'Rule not applicable: No move event with pick-up or Shipment-request move type found',
```

Reply:
[feature/update-same-source-and-pick-up-address 1f4cd19] chore: update the same-source-and-pick-up-address rule